master/buildbot/steps/trigger.py

```diff
@@ -89,11 +89,20 @@ def __init__(self, schedulerNames=None, sourceStamp=None, sourceStamps=None,
         BuildStep.__init__(self, **kwargs)
 
     def interrupt(self, reason):
-        # FIXME: new style step cannot be interrupted anymore by just calling finished()
-        #        (which was a bad idea in the first place)
-        # we should claim the self.brids, and CANCELLED them
-        # if they were already claimed, stop the associated builds via data api
-        # then the big deferredlist will automatically be called
+        # We claim the self.brids and CANCEL them.
+        # Sending cancel on a buildrequest through the data api,
+        # makes sure that if the requests were already claimed, i.e.
+        # the builds were started,
+        # a stop message will be sent on the messsage bus.
+        # However, even though the stop messages are sent, note that
```

Comment:
I dont see how this can happen. If waitForFinish is set to True, then the parent should wait for buildrequest finished event.